hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeHealthCheckerService.java

```diff
@@ -54,22 +58,35 @@ protected void serviceInit(Configuration conf) throws Exception {
    * @return the reporting string of health of the node
    */
   String getHealthReport() {
+    String healthReport = "";
```

Comment:
This would be a bit cleaner with a Joiner:

String scriptReport = (nodeHealthScriptRunner == null) ? null : nodeHealthScriptRunner.getHealthReport();
String discReport = dirsHandler.getDisksHealthReport(false);
String exceptionReport = nodeHealthException == null ? null : nodeHealthException.getMessage();
String healthReport = Joiner.on(SEPARATOR).skipNulls().join(scriptReport, discReport.equals("") ? null : discReport, exceptionReport);

The discReport throws a monkey wrench in the works because it's returning "" instead of null.  There's probably a more elegant solution that what I did above...